pages/add/add.jsx

```diff
@@ -1,12 +1,201 @@
 import React from 'react';
+import { browserHistory } from 'react-router';
+import HintMessage from '../../components/hint-message/hint-message.jsx';
+import Service from '../../js/service.js';
+import UserData from '../../js/user-data.js';
+import env from "../../config/env.generated.json";
+
+const CreatorInputField = () => {
+  return (
+    <input name="creators" type="text" placeholder="Name" width="150" className="form-control" />
+  );
+};
 
 export default React.createClass({
+  getInitialState() {
+    return {
+      numCreatorFields: 1,
+      loggedInStatusChecked: false
+    };
+  },
+  componentWillMount() {
+    console.log(`UserData.userLoggedInStatus.get() = `, UserData.userLoggedInStatus.get());
+  },
+  componentDidMount() {
+    let loggedInParam = this.props.router.location.query.loggedin;
+
+    if ( !loggedInParam || loggedInParam.toString().toLowerCase() === `true` ) {
+      // if there's no "loggedin" param presented in the URL OR if "loggedin" param is true
+      // we want to verify if user is really logged in by making a /userstatus call to Pulse API
+      this.verifyLoggedInStatus();
+    } else if ( loggedInParam && loggedInParam.toString().toLowerCase() === `false` ) {
+      // if "loggedin" param is presented in the URL and it is set to false
+      // we assume that user cannot be logged in using his/her Google Account
+      // (i.e., not using a Mozilla Staff account)
+      this.updateLoggedInInfo(false);
+    }
+  },
+  verifyLoggedInStatus() {
```

Comment:
this function show not exist here, this is "no related to the component in the slightest" functionality and so should just be in its own lib